dump/pldm_utils.cpp

```diff
@@ -1,93 +1,174 @@
 // SPDX-License-Identifier: Apache-2.0
 
+#include "pldm_utils.hpp"
+
 #include "xyz/openbmc_project/Common/error.hpp"
 
 #include <fmt/core.h>
 #include <libpldm/base.h>
 #include <libpldm/pldm.h>
+#include <libpldm/transport.h>
+#include <libpldm/transport/mctp-demux.h>
+#include <poll.h>
 
 #include <phosphor-logging/elog-errors.hpp>
 #include <phosphor-logging/log.hpp>
 
 namespace openpower::dump::pldm
 {
+
 using namespace phosphor::logging;
-namespace internal
+using NotAllowed = sdbusplus::xyz::openbmc_project::Common::Error::NotAllowed;
+using Reason = xyz::openbmc_project::Common::NotAllowed::REASON;
+pldm_instance_db* pldmInstanceIdDb = NULL;
+struct pldm_transport* pldmTransport = NULL;
+struct pldm_transport_mctp_demux* mctpDemux = NULL;
+
+PLDMInstanceManager::PLDMInstanceManager()
 {
-std::string getService(sdbusplus::bus::bus& bus, const std::string& path,
-                       const std::string& interface)
+    initPLDMInstanceIdDb();
+}
+
+PLDMInstanceManager::~PLDMInstanceManager()
 {
-    using namespace phosphor::logging;
-    constexpr auto objectMapperName = "xyz.openbmc_project.ObjectMapper";
-    constexpr auto objectMapperPath = "/xyz/openbmc_project/object_mapper";
+    destroyPLDMInstanceIdDb();
+}
 
-    auto method = bus.new_method_call(objectMapperName, objectMapperPath,
-                                      objectMapperName, "GetObject");
+void initPLDMInstanceIdDb()
```

Comment:
mixing c++ with C functions, why can't we have initPLDMInstanceIdDb, destroyPLDMInstanceIdDb  as private methods of the PLDMInstanceManager